borg/helpers.py

```diff
@@ -1040,17 +1040,33 @@ def __init__(self, total, step=5, start=0, same_line=False, msg="%3.0f%%", file=
         :param start: at which percent value to start
         :param same_line: if True, emit output always on same line
         :param msg: output message, must contain one %f placeholder for the percentage
-        :param file: output file, default: sys.stderr
         """
         self.counter = 0  # 0 .. (total-1)
         self.total = total
         self.trigger_at = start  # output next percentage value when reaching (at least) this
         self.step = step
-        if file is None:
-            file = sys.stderr
-        self.file = file
         self.msg = msg
         self.same_line = same_line
+        self.handler = None
+        self.logger = logger.getChild('progress')
+
+        # If there are no handlers, set one up explicitly because the
+        # terminator and propagation needs to be set.  If there are,
+        # they must have been set up by BORG_LOGGING_CONF: skip setup.
+        if not self.logger.handlers:
+            self.handler = logging.StreamHandler(stream=sys.stderr)
+            self.handler.setLevel(logging.INFO)
+            self.handler.terminator = '\r' if self.same_line else '\n'
```

Comment:
I did test.  StreamHandler isn't documented to flush after every message, but that's what the implementation does.  One of the tests added earlier looks for the carriage return in stderr after the progress message was sent, but I modified it to send two progress messages between checks.  I can't think of another way to test that behavior in case StreamHandler changes behavior.

I could subclass StreamHandler and add a flush at the end of .emit() after calling super().emit() (which would currently double flush), but that seems excessive to me.